webui/src/__tests__/common.js

```diff
@@ -24,25 +24,47 @@ if (!USER_NAME || !USER_PASS || !USER_EMAIL) {
     console.log("username or password or email environment variables not defined!");
 }
 
-export function step(msg) {
-    console.log(msg, "...");
+export async function getBodyText() {
+    return document.body.innerText.trim();
 }
 
-export function print_obj(o) {
-    for (var p in o) {
-        console.log(p);
-    }
+export async function getElementText(selector) {
+    return document.querySelector(selector).innerText.trim();
 }
 
-export async function assertPageContains(page, text) {
-    let body = await page.evaluate(()=>document.body.textContent);
-    await expect(body).toContain(text);
+export async function getElementsJoinedText(selector) {
+    const ret = [];
+    for (var el in document.querySelectorAll(selector)) {
+        ret.push(el.innerText.trim());
+    }
+    return ret.join("\n")
 }
 
-export async function assertElementText(page, selector, text) {
-    const fn = (selector) => {
-        document.querySelector(selector).innerText
-    };
-    let elementText = await page.evaluate(fn, selector);
-    return expect(elementText == text);
+export async function login(page) {
+    console.log("login started");
+    expect(page).toBeInstanceOf(Nightmare);
+
+    await page.wait('#header-navbar-collapse > .user > li > a')
+            .click('#header-navbar-collapse > .user > li > a');
+
+    const url = await page.url();
+    expect(url).toEqual(B2ACCESS_AUTH_URL);
+
+    await page.wait('#AuthenticationUI\\.username');
+    const body = await page.evaluate(getBodyText);
+    expect(body).toContain("Login to UNITY OAuth2 Authorization Server");
+
+    await page.type('#AuthenticationUI\\.username', USER_NAME)
+            .type('#WebPasswordRetrieval\\.password', USER_PASS)
+            .click('#AuthenticationUI\\.authnenticateButton')
+            .wait('#IdpButtonsBar\\.confirmButton')
+            .click('#IdpButtonsBar\\.confirmButton')
+            .wait('#header-navbar-collapse > .user > li > a');
+
+    expect(await page.title()).toEqual('B2SHARE');
+
+    const emailText = await page.evaluate(getElementText, '#header-navbar-collapse > .user > li > a');
+    expect(emailText).toEqual(USER_EMAIL);
```

Comment:
Using the "show: true" option it can successfully authenticate and shows my email address on the corner of the page, but returns this error on console: Expected value to equal:
      "sarahba@kth.se"
    Received:
      "Login"

Also, after running the test for a couple of times, it stuck on the Unity page and doesn't go forward to the B2share page, the same issue as before.